register/forms.py

```diff
@@ -23,22 +20,25 @@ def save(self, application):
 
     def update_forms(self):
         forms = self.fetch()
+        if not forms:
+            return []
         applications = map(self.map, forms)
-        return map(self.save, applications)
+        return [app.save() for app in applications]
 
 
 class TypeformFetcher(ApplicationFormFetcher):
     base_url = 'https://api.typeform.com/v1/form/'
 
-    def get_form_id(self):
-        return settings.REGISTER_APP['applications_form_id']
+    @property
```

Comment:
It's an option to define abstract methods in a class. If the method is called gives NotImplemented. Following code style from Django Rest Framework. 

From: https://docs.python.org/2/library/exceptions.html

`In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method.`